File: examples/CBuildr4/exdrv0.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "ExDrv0.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "OvcBase"
#pragma link "OvcCmbx"
#pragma link "OvcDlg"
#pragma link "OvcDRpVw"
#pragma link "OvcRptVw"
#pragma link "OvcRvCbx"
#pragma link "OvcRVIdx"
#pragma link "OvcRvPDg"
#pragma link "ovcbase"
#pragma link "ovccmbx"
#pragma link "ovcdlg"
#pragma link "ovcdrpvw"
#pragma link "ovcrptvw"
#pragma link "ovcrvcbx"
#pragma link "ovcrvidx"
#pragma link "ovcrvpdg"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
    : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::SpeedButton1Click(TObject *Sender)
{
  OvcRvPrintDialog1->Execute();
}
//---------------------------------------------------------------------------

void __fastcall TForm1::OvcDataReportView1Filter(TObject *Sender,
      TOvcDataRvItem *Data, int FilterIndex, bool &Include)
{
  if (FilterIndex == 0)
    Include = Data->AsBoolean[4];
}
//---------------------------------------------------------------------------
 

File: examples/Delphi/EXDRV0.pas
{$J+} {Writable constants}

unit ExDrv0;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, OvcBase, OvcRVIdx, OvcRptVw, OvcDRpVw, StdCtrls, ExtCtrls,
  OvcSplit, OvcCmbx, OvcRvCbx;

type
  TForm1 = class(TForm)
    OvcDataReportView1: TOvcDataReportView;
    OrController1: TOvcController;
    Panel1: TPanel;
    OvcViewComboBox1: TOvcViewComboBox;
    OvcSplitter1: TOvcSplitter;
    OvcDataReportView2: TOvcDataReportView;
    Panel2: TPanel;
    OvcViewComboBox2: TOvcViewComboBox;
    procedure OvcDataReportView1DragOver(Sender, Source: TObject; X,
      Y: Integer; State: TDragState; var Accept: Boolean);
    procedure OvcDataReportView1DragDrop(Sender, Source: TObject; X,
      Y: Integer);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var                            
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.OvcDataReportView1DragOver(Sender, Source: TObject; X,
  Y: Integer; State: TDragState; var Accept: Boolean);
begin
  Accept := (Source <> Sender)
    and (Source is TOvcDataReportView)
    and (TOvcDataReportView(Source).CurrentItem <> nil);
end;

procedure TForm1.OvcDataReportView1DragDrop(Sender, Source: TObject; X,
  Y: Integer);
begin
  if TOvcDataReportView(Source).CurrentItem <> nil then
    if Source <> Sender then begin
      TOvcDataReportView(Sender).Items.Add.Assign(
        TOvcDataReportView(Source).CurrentItem);
      TOvcDataReportView(Source).CurrentItem.Free;
    end;
end;

end.
                             
